crates/p2p-network/examples/p2p.rs

```diff
@@ -0,0 +1,54 @@
+//! To be run manually to test auto-reconnect.
+
+use bytes::Bytes;
+use futures_util::StreamExt;
+use gw_config::P2PNetworkConfig;
+use gw_p2p_network::*;
+use tentacle::{builder::MetaBuilder, multiaddr::Protocol, service::ProtocolMeta, ProtocolId};
+
+const PROTOCOL_ZERO: ProtocolId = ProtocolId::new(0);
+
+fn protocol() -> ProtocolMeta {
+    MetaBuilder::new()
+        .id(PROTOCOL_ZERO)
```

Comment:
https://github.com/nervosnetwork/axon/blob/main/core/network/src/protocols/mod.rs#L92

Consider using compression of messages to save bandwidth usage, with relatively high gains on large messages

Reply:
Noted.

I think I will add compression (and authentication) when we allow other nodes to connect from the internet. Right now it's just for internal nodes.